nft-marketplace/src/auction.rs

```diff
@@ -177,26 +152,40 @@ impl Market {
 
         check_attached_value(item.ft_contract_id, price);
 
-        if price <= auction.current_price {
+        let previous_price = auction.current_price;
+        let previous_winner = auction.current_winner;
+
+        if price <= previous_price {
             panic!("Cant offer less or equal to the current bid price")
         }
 
+        if auction.ended_at <= exec::block_timestamp() + auction.bid_period {
+            auction.ended_at = exec::block_timestamp() + auction.bid_period;
+        }
```

Comment:
For example, the auction ends at 6 pm and `bid_period` is 5 minutes. If bids were made before 5.55 pm then auction is not extended. But if someone bets at 5.59 then auction extended to 18.04 
so that the rest of the participants have time to see the new bet and possibly make another one.